pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/JavaNode.java

```diff
@@ -89,6 +89,23 @@ default JavaNode getLastChild() {
 
     GenericToken jjtGetLastToken();
 
+
+    /**
+     * Returns the node representing the type declaration this node is
+     * found in. The type of that node is the type of the {@code this}
+     * expression.
+     *
+     * <p>This returns null for nodes that aren't enclosed in a type declaration.
+     * This includes {@linkplain ASTPackageDeclaration PackageDeclaration},
+     * This includes {@linkplain ASTImportDeclaration ImportDeclaration},
+     * {@linkplain ASTCompilationUnit CompilationUnit}, and top-level
+     * {@linkplain ASTAnyTypeDeclaration AnyTypeDeclaration}s.
```

Comment:
I think it will return null for anything in modules as well (ASTModuleDeclaration, ...)

Reply:
Ah yes that's true!

Reply:
Fixed with deab05f3b8637d2de028878af0ff77bdb439286c